### PR TITLE
Feat(Query): Memory Not Limited for Docker Compose

### PR DESCRIPTION
**Pull request by cxAndreFelicidade**
**Proposed Changes**
- Query: Memory Not Limited for Docker Compose

I submit this contribution under the Apache-2.0 license.

### CODE REVIEW

File: assets/queries/dockerCompose/memory_not_limited/metadata.json [hunk 1]
```diff
@@ -0,0 +1,10 @@
+{
+    "id": "bb9ac4f7-e13b-423d-a010-c74a1bfbe492",
+    "queryName": "Memory Not Limited",
+    "severity": "MEDIUM",
+    "category": "Resource Management",
+    "descriptionText": "Check if theres is a memory limit for containers in the docker compose document.",
```
Comment (rafaela-soares) [inline assets/queries/dockerCompose/memory_not_limited/metadata.json:6]:
```suggestion
    "descriptionText": "Memory limits should be defined for each container. This prevents potential resource exhaustion by ensuring that containers consume not more than the designated amount of memory",
```

File: assets/queries/dockerCompose/memory_not_limited/query.rego [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+	resource := input.document[i]
+    version := resource.version
+    to_number(version) >= 3
+	service_parameters := resource.services[name]
+   	limits := service_parameters.deploy.resources.limits
+    not common_lib.valid_key(limits, "memory")
+   
+	result := {
+		"documentId": sprintf("%s", [resource.id]),
+		"searchKey": sprintf("services.%s",[name]),
```
Comment (rafaela-soares) [inline assets/queries/dockerCompose/memory_not_limited/query.rego:16]:
```suggestion
        "searchKey": sprintf("services.%s.deploy.resources.limits",[name]),
```

File: assets/queries/dockerCompose/memory_not_limited/query.rego [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+		"issueType": "IncorrectValue",
+		"keyExpectedValue": "There is a memory limit in deployed resources",
+		"keyActualValue": "There is no memory limit in deployed resources",
+		"searchLine": common_lib.build_search_line(["services", name], []),
```
Comment (rafaela-soares) [inline assets/queries/dockerCompose/memory_not_limited/query.rego:20]:
```suggestion
		"searchLine": common_lib.build_search_line(["services", name, "deploy", "resources", "limits"], []),
```

**Review by rafaela-soares — CHANGES_REQUESTED** [review]

**Review by joaorufi — DISMISSED** [review]

**Review by joaorufi — APPROVED** [review]

File: assets/queries/dockerCompose/memory_not_limited/query.rego [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+    to_number(version) >= 3
+	service_parameters := resource.services[name]
+   	limits := service_parameters.deploy.resources.limits
+    not common_lib.valid_key(limits, "memory")
+   
+	result := {
+		"documentId": sprintf("%s", [resource.id]),
+		"searchKey": sprintf("services.%s.deploy.resources.limits",[name]),
+		"issueType": "IncorrectValue",
+		"keyExpectedValue": "'deploy.resources.limits.memory' is defined",
```
Comment (joaorufi) [inline assets/queries/dockerCompose/memory_not_limited/query.rego:18]:
```suggestion
		"issueType": "MissingAttribute",
		"keyExpectedValue": "'deploy.resources.limits.memory' to be defined",
```

File: assets/queries/dockerCompose/memory_not_limited/query.rego [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+    version := resource.version
+    to_number(version) >= 3
+	service_parameters := resource.services[name]
+    not common_lib.valid_key(service_parameters.deploy, "resources")
+   
+	result := {
+		"documentId": sprintf("%s", [resource.id]),
+		"searchKey": sprintf("services.%s.deploy",[name]),
+		"issueType": "IncorrectValue",
+		"keyExpectedValue": "'deploy.resources.limits.memory' is defined",
```
Comment (joaorufi) [inline assets/queries/dockerCompose/memory_not_limited/query.rego:52]:
```suggestion
		"issueType": "MissingAttribute",
		"keyExpectedValue": "'deploy.resources' to be defined",
```

**Review by joaorufi — CHANGES_REQUESTED** [review]

**Review by rafaela-soares — APPROVED** [review]
LGTM 🚀

**Pull request merged**